### PR TITLE
#518 Apply new UI for login in mobile

### PR DESCRIPTION
**Pull request by hoangdat**
https://user-images.githubusercontent.com/6462404/167122649-4d18104b-ab08-4f45-abcd-1dcef144b89c.mp4

Other demo files are too big, I attached to chat channel

### CODE REVIEW

File: core/lib/presentation/utils/responsive_utils.dart [hunk 1]
```diff
@@ -51,4 +53,8 @@ class ResponsiveUtils {
       isMobile(context) ? _loginTextFieldWidthSmallScreen : _loginTextFieldWidthLargeScreen;
 
   double getWidthLoginButton() => _loginButtonWidth;
+
+  bool isHeightShortest(BuildContext context) {
```
Comment (dab246) [inline core/lib/presentation/utils/responsive_utils.dart:57]: What is this used for?

File: lib/features/login/presentation/login_controller.dart [hunk 2]
```diff
@@ -82,7 +98,7 @@ class LoginController extends GetxController {
 
   @override
   void onClose() {
-    urlInputController.dispose();
```
Comment (dab246) [inline lib/features/login/presentation/login_controller.dart:85]: Why delete it?

File: lib/features/login/presentation/login_view.dart [hunk 3]
```diff
@@ -1,116 +1,182 @@
...
+  final _responsiveUtils = Get.find<ResponsiveUtils>();
   final keyboardUtils = Get.find<KeyboardUtils>();
 
   LoginView({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      resizeToAvoidBottomInset: false,
+      resizeToAvoidBottomInset: true,
```
Comment (dab246) [inline lib/features/login/presentation/login_view.dart:20]: This property defaults to `true`, so you don't need to add it here.

File: lib/features/login/presentation/login_view.dart [hunk 4]
```diff
@@ -125,40 +191,78 @@ class LoginView extends GetWidget<LoginController> {
...
-            shape: MaterialStateProperty.all(RoundedRectangleBorder(
-              borderRadius: BorderRadius.circular(80),
-              side: const BorderSide(width: 0, color: AppColor.buttonColor)))),
-          child: Text(AppLocalizations.of(context).login, style: const TextStyle(fontSize: 16, color: Colors.white)),
-          onPressed: () {
-            keyboardUtils.hideKeyboard(context);
-            loginController.handleLoginPressed();
-          }),
+    return Container(
+      margin:  const EdgeInsets.only(bottom: 16, left: 24, right: 24),
```
Comment (dab246) [inline lib/features/login/presentation/login_view.dart]: Excess white space

File: lib/features/login/presentation/widgets/login_text_input_builder.dart [hunk 5]
```diff
@@ -0,0 +1,196 @@
+// LinShare is an open source filesharing software, part of the LinPKI software
```
Comment (dab246) [inline lib/features/login/presentation/widgets/login_text_input_builder.dart:1]: Remove this doc

File: lib/features/login/presentation/base_login_view.dart [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+import 'package:get/get.dart';
+import 'package:tmail_ui_user/features/login/domain/state/authentication_user_state.dart';
+import 'package:tmail_ui_user/features/login/presentation/login_controller.dart';
+import 'package:tmail_ui_user/features/login/presentation/login_form_type.dart';
+import 'package:tmail_ui_user/features/login/presentation/state/login_state.dart';
+import 'package:tmail_ui_user/features/login/presentation/widgets/login_input_decoration_builder.dart';
+import 'package:tmail_ui_user/features/login/presentation/widgets/login_text_input_builder.dart';
+import 'package:tmail_ui_user/main/localizations/app_localizations.dart';
+
+abstract class BaseLoginView extends GetWidget<LoginController> {
```
Comment (dab246) [inline lib/features/login/presentation/base_login_view.dart:16]: good idea

File: lib/features/login/presentation/login_view_web.dart [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+                  ),
+                  Padding(
+                    padding: const EdgeInsets.only(top: 16),
+                    child: (SloganBuilder(arrangedByHorizontal: true, )
+                      ..setLogoSVG(imagePaths.icEncrypted)
+                      ..setSizeLogo(48.0)
+                      ..setPadding(const EdgeInsets.only(left: 12))
+                      ..setSloganText(AppLocalizations.of(context).encryptedMailbox)
+                      ..setSloganTextStyle(const TextStyle(fontSize: 24, fontWeight: FontWeight.w400, color: AppColor.colorNameEmail)))
+                        .build()
```
Comment (dab246) [inline lib/features/login/presentation/login_view_web.dart:108]: too indented

File: lib/features/login/presentation/login_view_web.dart [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+                  ),
+                  Padding(
+                    padding: const EdgeInsets.only(top: 16),
+                    child: (SloganBuilder(arrangedByHorizontal: true, )
+                      ..setLogoSVG(imagePaths.icTeam)
+                      ..setSizeLogo(48.0)
+                      ..setPadding(const EdgeInsets.only(left: 12))
+                      ..setSloganText(AppLocalizations.of(context).manageEmailAsATeam)
+                      ..setSloganTextStyle(const TextStyle(fontSize: 24, fontWeight: FontWeight.w400, color: AppColor.colorNameEmail)))
+                        .build()
```
Comment (dab246) [inline lib/features/login/presentation/login_view_web.dart:118]: same

File: lib/features/login/presentation/login_view_web.dart [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+                  Padding(
+                    padding: const EdgeInsets.only(top: 16),
+                    child: (SloganBuilder(arrangedByHorizontal: true, )
+                      ..setLogoSVG(imagePaths.icIntegration)
+                      ..setSizeLogo(48.0)
+                      ..setPadding(const EdgeInsets.only(left: 12))
+                      ..setSloganText(AppLocalizations.of(context).multipleIntegrations)
+                      ..setSloganTextStyle(const TextStyle(fontSize: 24, fontWeight: FontWeight.w400, color: AppColor.colorNameEmail)))
+                        .build()
+                  ),
```
Comment (dab246) [inline lib/features/login/presentation/login_view_web.dart:129]: same

**Review by johnnguyenit39 — APPROVED** [review]

File: core/lib/presentation/utils/responsive_utils.dart (hunk 1 above)
Comment (hoangdat) [reply core/lib/presentation/utils/responsive_utils.dart:57]: use for padding the top in case the screen have smaller than 600

**Comment by chibenwa** [comment]
Is it intendeed not to have the logo on the login page? 

Or am I missing something?

**Review by dab246 — APPROVED** [review]
LBTM. Please rebase code

**Pull request merged**